### PR TITLE
5 establishing connection with database

### PR DESCRIPTION
**Pull request by hubertborysowski**
# Read me
Establishment with database works fine.

`For tests only` sections are still there, so you can check if everything works fine on your devices.
Health check is made by `catch` method.

**NOTE:** The program seeks database called `test` with table `students` in it. Change it to desired one or create temporary database with specified table (of course remember to fill this table with some data). Program seeks for int, string, and int again.

### CODE REVIEW

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: This method should be documented. Other than that, the name is not self-explanatory so I'm not exactly sure what is it that this method is doing

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: Instead of including tests in the source code, create separate tests for that. In order to do that, you need to mock the behavior of the database to see if the application behaves as expected depending on whether the DB is working properly or not.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: Avoid using magic values. Use a static variable at the beginning of the file or remove this line altogether.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: Avoid using magic values

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: Avoid using magic values. Use a static variable at the beginning of the file

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: Avoid using magic values. This line is not readable, nor self-explanatory. Looking at it, it is not possible to determine, what data will be printed. Try to make it cleaner, e.g. by assigning elements of the `resultSet` to separate variables. Other than that, are you sure that this line will work if the query will be changed?

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplication.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplication.java]: This class handles the app logic. Depending on what the method is meant to be doing (it is not possible to figure this out by looking at it), it might be at a right spot

**Review by Mateusz-Dobrzynski — CHANGES_REQUESTED** [review]
You need to clean up and document your code, as well as create tests for it

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (hubertborysowski) [reply src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: It is impossible not to use catch function in this form. Even [official documentation](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-usagenotes-connect-drivermanager.html) shows this kind of practice.

Comment (hubertborysowski) [reply src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: SQL demands usage of this practice as the connection establishment enforces that.

File: src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java
Comment (Mateusz-Dobrzynski) [inline src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java]: This test does not make any assertions and as such doesn't test anything. Check several test cases. Mock the database behavior if you have to.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (Mateusz-Dobrzynski) [reply src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: My comment does not refer to any of the above but to the magic value you used. Refrain from using them, if possible.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java]: You don't have to explain what the methods you're using do. They are documented by their authors. What matters is the documentation of your own code and what exactly it does.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java]: This comment is redundant as the name of the method speaks for itself. Try to describe its behavior in more detail.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java]: This code doesn't seem po perform any particular function and as such shouldn't be included in the project. You can move it to the test class if you insert the said `operations`

**Review by Mateusz-Dobrzynski — CHANGES_REQUESTED** [review]
Do not include the generated documentation just yet. What matters are the comments in the source code (which, incidentally speaking, could be more detailed). Including the generated docs right now will make code review more difficult, not easier.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/mysqlConnection.java]: Do not include credentials in the source code! Create a separate config file for that and DO NOT include its contents in the repository

File: src/test/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabaseTest.java
Comment (Mateusz-Dobrzynski) [inline src/test/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabaseTest.java:1]: This unit test is device-dependent (it requires a functional database) and as such does not serve its purpose. Other than that, it does not make any assertions. As stated in my previous reviews, create several test cases and mock the database behavior.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java]: Avoid using magic values and **under no circumstances** include credentials in the source code. Create a separate config file for that and **do not** include it in your commits (include the file in `.gitignore`. Include instructions in the README file on where to enter the credentials.

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java]: Unused variables

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java
Comment (Mateusz-Dobrzynski) [inline src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java]: Unused connection. It is immediately closed and as such does not serve the application

File: src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java
Comment (Mateusz-Dobrzynski) [inline src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java:1]: This file is redundant. Turn `EstablishConnectionWithDatabaseTest.java` into a useful test instead.

**Review by Mateusz-Dobrzynski — CHANGES_REQUESTED** [review]

File: src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java
Comment (hubertborysowski) [reply src/main/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabase.java]: So you want this to receive (save in some kind of an array) and send data to the database?

**Comment by hubertborysowski** [comment]
Unused connection is still present, same as variables, waiting for response for previous comment about that.
Useless test is still useless. Trying to work on it, but having problems with mocking database.

**Comment by hubertborysowski** [comment]
Not ready for merging.
WIP.

**Comment by hubertborysowski** [comment]
Works! 🤖
Waiting for review 📑

File: src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java
Comment (Mateusz-Dobrzynski) [inline src/test/java/org/lookout_studios/meals_management_system/meals_management_system/MealsManagementSystemApplicationTests.java:16]: You don't need to invoke the tests in a separate class. With these lines you test the same code twice: once directly in `EstablishConnectionWithDatabaseTest` and here again, invoking the same test methods. You can delete these lines.

File: src/test/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabaseTest.java
Comment (Mateusz-Dobrzynski) [inline src/test/java/org/lookout_studios/meals_management_system/meals_management_system/EstablishConnectionWithDatabaseTest.java:13]: Document this class by explaining what it does

**Review by Mateusz-Dobrzynski — APPROVED** [review]
With mocks present, the code is tested and as such can be merged. Consider other comments before merging

**Pull request merged**
